### DISCUSSION TITLE
Support the Filipino Language

### DISCUSSION THREAD

**Discussion by LeNitrous** [ppy/osu#18851] [Ideas] [answered]
Through community efforts of the osu!philippines community, we are keeping the Filipino language up to date for both web and lazer. The Filipino language is already available in resources but is being unused. This is an example of it being used right now.
![image](https://user-images.githubusercontent.com/20495991/175820973-9d303dc6-1381-44e5-a717-60669d6cd4ef.png)

.NET checks available cultures through the user's system. As Filipino (Philippines) is not available on Windows upon installation, users have to install the Filipino Language Pack through the Control Panel (Settings App in Windows 11). The same can possibly be said for other platforms.

Additionally, The language uses the ISO code `fil-PH` (used by Crowdin and Windows)
![image](https://user-images.githubusercontent.com/20495991/175821149-3d6a6d08-89b3-427a-b776-02d28c2bcf69.png)

The web side of things seem to not also make the language an available option. A separate issue or discussion will be made on osu web for this matter as well.

**Comment by peppy** [comment]
Thanks for the bump. I check on language completion every once in a while but haven't done so as of recent.

I'll take a look this week and add it to both projects if things look good.

**Comment by peppy** [comment]
@LeNitrous how did you get this working in-game? I've tried but the strings just don't show up, even though everything looks correct.

Branch with my changes at https://github.com/ppy/osu/compare/master...peppy:osu:add-language-tagalog?expand=1

**Reply by LeNitrous** [reply]
Just checked out your changes and it just works fine.
![image](https://user-images.githubusercontent.com/20495991/176125137-74ac6a7d-5435-4415-9b4a-1aab4d5f4069.png)

**Reply by LeNitrous** [reply]
Additionally, it should be referred to as "Filipino" as stated in [law](https://lawphil.net/statutes/repacts/ra1991/ra_7104_1991.html). "Tagalog" is one of the native languages spoken here in the Philippines and it is where "Filipino" derives itself from.

Additionally, websites like Facebook and Google use "Filipino"

**Reply by bdach** [reply]
re: language name - see https://github.com/ppy/osu/pull/13572#issuecomment-864369298

**Comment by bdach** [comment]
@peppy I believe I know why this may not be working. This is most likely msbuild's fault, at least it is on my machine.

The long and short of it is that msbuild isn't copying the `osu.Game.Resources.resources.dll` satellite assembly for `fil-PH`. This manifests in the fact that when I build on windows, there is no `fil-PH` directory in `bin/Debug/net6.0`.

However, when I go to my nuget cache at `%USERPROFILE%\.nuget\packages\ppy.osu.game.resources\2022.628.0\lib\netstandard2.1\fil-PH`, the satellite assembly is correctly present.

The working theory is that msbuild, being likely written back in .NET Framework, is somehow using the "old" pathway of querying cultures and is not aware that it should copy the `fil-PH` satellite assembly until the Filipino language pack is installed system-wide, which presumably adds the culture to the old proper registry pathways and allows msbuild to pick up the satellite assembly.

If you manually copy the `fil-PH` folder from the resources package site to the build output directory, the localisations will work with no actual code changes on top of your branch. At least they do here.

**Reply by peppy** [reply]
Nice find.

**Reply by smoogipoo** [reply]
This sounds like a reoccurrence of the discoveries made in https://github.com/ppy/osu-localisation-analyser/issues/44, is that right?

**Reply by bdach** [reply]
Looks like the same issue to me, yeah.

**Reply by peppy** [reply]
Do we know the correct code to use in this case?

**Reply by smoogipoo** [reply]
I've tried `fil` and `fil-Latn` and neither seem to work :/ Not sure...

**Reply by bdach** [reply]
Those two are the only ones I'd ever expect to work, as per https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c?redirectedfrom=MSDN

**Comment by Joehuu** [comment]
Note that this was a nuget issue and was fixed since https://learn.microsoft.com/en-us/nuget/release-notes/nuget-6.13:
> Satellite assemblies for three-letter languages are not copied from NuGet package - [#12253](https://github.com/NuGet/Home/issues/12253)

But that fix is only packaged on .NET 9/10, unless the tooling uses nuget separately, but I believe that is waste of time to get working than just upgrading to .NET 9/10.

**Reply by Joehuu** [reply, marked as answer]
https://github.com/ppy/osu/pull/38779
